engine/src/main/java/io/zeebe/engine/processor/workflow/timer/TriggerTimerProcessor.java

```diff
@@ -123,27 +130,30 @@ private boolean shouldReschedule(final TimerRecord timer) {
 
   private void rescheduleTimer(
       final TimerRecord record, final TypedStreamWriter writer, final ExecutableCatchEvent event) {
-    if (event.getTimer() == null) {
+    final Timer timer;
+    try {
+      timer = event.getTimerFactory().apply(expressionProcessor, record.getElementInstanceKey());
+    } catch (Exception e) {
       final String message =
           String.format(
               "Expected to reschedule repeating timer for element with id '%s', but no timer definition was found",
               BufferUtil.bufferAsString(event.getId()));
-      throw new IllegalStateException(message);
+      throw new IllegalStateException(message, e);
```

Comment:
Instead of throwing an exception, we should create an incident if the evaluation fails. However, this can be done in a follow-up issue.

Reply:
@saig0 The original implementation already throws an Exception. Is that a bug? Can you explain the behaviour differences between an exception and an incident in this situation?

Reply:
An incident is created to indicate a problem to the user and it can be resolved when the problem is solved by the user. 

If an exception is thrown in this processor then the whole workflow instance is added to the "backlist". No further event/command is applied to this instance. It can not be repaired.

Previously, the exception would be only thrown if there was a bug in the processor or state - it was just to prevent an NPE. But now, an evaluation could fail on rescheduling if a variable has changed. 

Saying that I realize that the incident could not be resolved by the current implementation :thinking: It just works for jobs or workflow instance-related events /https://github.com/zeebe-io/zeebe/blob/develop/engine/src/main/java/io/zeebe/engine/processor/workflow/incident/ResolveIncidentProcessor.java#L69).

So, we have two options: don't evaluate the expression or extend the incident logic. I would prefer the first option for now. Let's try to avoid throwing an exception if possible.

Reply:
See #4208